### ISSUE TITLE
[ENH] hierarchical reconciliation - optional removal of sum/dunder levels

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#4767]
Currently, any application of reconciliation adds dunder levels `__total` etc if they are not there.

It should be possible though to apply reconciliation and not have the dunder levels added.

An example would be applying shrinkage or top-down reconciliation to a hierarchy, and then removing the "sum" levels. This would result in a non-trivially different forecast, even if the total levels are not present at the end.

Therefore, I would suggest:

* a transformer that removes the "total" levels. This could simply be the inverse of the `Aggregator`, or a new transformer. If part of `Aggregator`, whether the inverse is skipped could be controlled by a parameter of `Aggregator`.
* the `Reconciler` and `ReconcilerForecaster` should have an option whether the "total" levels are preserved, or removed at the end.

FYI @ciaran-g, @VyomkeshVyas - what do you think?

**Comment by ciaran-g** [comment]
Hi @fkiraly - not sure how common the use-case would since there would be no requirement for the forecasts to sum together appropriately. But in principle yeah I agree that it is an option and the suggestions you make sound like a good way to get there

**Comment by fkiraly** [comment]
> since there would be no requirement for the forecasts to sum together appropriately

Yes - my thinking was, it could still improve overall accuracy of forecasts. Imagining the scenario where all sub-level values are always a fixed percentage of the top-level total, for instance, plus noise. In this case, the top-down method will provide better forecasts measured on the individual level, even if the top level is not explicitly there in the data frame.

**Comment by ciaran-g** [comment]
Nice one, yeah I agree 👍

**Comment by felipeangelimvieira** [comment]
This issue was fixed in #7127
